example-apps/demo/source/components/router-toy/data-2.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'router-toy-data-2',
+  template: `
+  <div>
+    <h3>Hello There!!</h3>
+    <h4>Message: {{message}}</h4>
+    <h4>Name: {{name}}</h4>
+  </div>
+  `
+})
+export default class RouterToyData2 {
+  public message: string;
+  public name: string;
+  private params$: any;
+
+  constructor(private activatedRoute: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.params$ = this.activatedRoute.params.subscribe(({name, message}) => {
```

Comment:
Maybe, but for now we don't really have any functionality which inspects that, this is basically just here so that the router links don't look stupid.